threedi_modelchecker/simulation_templates/laterals/dwf_calculation.py

```diff
@@ -0,0 +1,229 @@
+             AND impsurf.id = impmap.impervious_surface_id GROUP BY impsurf.id),
+        inhibs_per_node AS (
+            SELECT impmap.impervious_surface_id, impsurfcount.nr_of_inhabitants,
+                   impmap.connection_node_id, impsurfcount.dry_weather_flow, impsurfcount.weighted_flow
+            FROM imp_surface_count impsurfcount, v2_impervious_surface_map impmap
+            WHERE impsurfcount.id = impmap.impervious_surface_id)
+        SELECT ipn.connection_node_id, SUM(ipn.nr_of_inhabitants), SUM(ipn.weighted_flow)
+        FROM inhibs_per_node ipn GROUP BY ipn.connection_node_id
+        """
+    ):
+        connection_node_id, nr_of_inhabitants_sum, weighted_flow_sum = row
+        # DWF per person example: 120 l/inhabitant / 1000 = 0.12 m3/inhabitant
+        dwf_per_node = (
+            nr_of_inhabitants_sum * (weighted_flow_sum / nr_of_inhabitants_sum) / 1000
+        )
+        dwf_per_node_per_second.append([connection_node_id, dwf_per_node / 3600])
+
+    conn.close()
+
+    return dwf_per_node_per_second
+
+
+def generate_dwf_lateral_json(
+    spatialite_filepath: str, start_time: str, duration: int, dwf_factors: Optional[str]
+) -> List[dict]:
+
+    dwf_on_each_node = read_dwf_per_node(spatialite_filepath)
+    dwf_factor_per_timestep = start_time_and_duration_to_dwf_factors(
+        start_time=start_time, duration=duration, dwf_factors=dwf_factors
+    )
+    # Initialize list that will hold JSON
+    dwf_list = []
+
+    # Generate JSON for each connection node
+    for dwf_node in dwf_on_each_node:
+        dwf_per_timestep = """"""
+        for row in dwf_factor_per_timestep:
+            dwf_per_timestep = (
+                dwf_per_timestep + str(row[0]) + "," + str(dwf_node[1] * row[1]) + "\n"
+            )
+
+        dwf_per_timestep = dwf_per_timestep[:-1]
+        dwf_list.append(
+            {
+                "offset": 0,
+                "interpolate": 0,
+                "values": dwf_per_timestep,
+                "units": "m3/s",
+                "connection_node": dwf_node[0],
+            }
+        )
+
+    return dwf_list
+
+
+def dwf_json_to_csv(dwf_list: List[dict], output_csv_file: str) -> None:
+    """Write DWF results to JSON."""
+    with open(output_csv_file, "w", newline="") as csv_file:
+        writer = csv.writer(csv_file)
+        writer.writerow(["id", "connection_node_id", "timeseries"])
+        for i, row in enumerate(dwf_list):
+            connection_node_id = row["connection_node"]
+            timeseries = row["values"]
+            writer.writerow([str(i), str(connection_node_id), timeseries])
+
+
+def str_to_seconds(time_str: str) -> int:
+    """Get Seconds from time."""
+    m, s = time_str.split(":")
+    return int(m) * 60 + int(s)
+
+
+class DWFCalculator:
+    """Calculate dry weather flow (DWF) from sqlite."""
+
+    def __init__(
+        self,
+        spatialite_path: str,
+        start_time: str,
```

Comment:
De vraag is waar deze waarden vandaan gaan komen.

`duration` is (geloof ik) een probleem, zie eerder comment.
`start_time` zit denk ik in de global_settings.